systemservices/resolver/resolver.go

```diff
@@ -0,0 +1,66 @@
+package resolver
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mesg-foundation/core/api"
+)
+
+// Resolvers tasks.
+const (
+	addPeersTask string = "AddPeers"
+	resolveTask         = "Resolve"
+)
+
+// Resolver is the system service responsible for getting the address of
+// other node of the network that are running the desired service.
+type Resolver struct {
+	api       *api.API
+	serviceID string
+}
+
+// New creates a new instance of the Resolver system service.
+func New(serviceID string, api *api.API) *Resolver {
+	return &Resolver{
+		api:       api,
+		serviceID: serviceID,
+	}
+}
+
+// AddPeers is the task that actually new add peers to the service.
+func (r *Resolver) AddPeers(addresses []string) error {
+	e, err := r.api.ExecuteAndListen(r.serviceID, addPeersTask, map[string]interface{}{
+		"addresses": addresses,
+	})
+	if err != nil {
+		return err
+	}
+
+	switch e.Output {
+	case "success":
+		return nil
+	case "error":
+		return errors.New(e.OutputData["error"].(string))
+	default:
+		return fmt.Errorf("unexpected output %s", e.Output)
+	}
+}
+
+// Resolve is the task that return the address of a core that runs the desired service.
+func (r *Resolver) Resolve(serviceID string) (address string, err error) {
+	e, err := r.api.ExecuteAndListen(r.serviceID, resolveTask, map[string]interface{}{
+		"serviceID": serviceID,
+	})
+
+	switch e.Output {
+	case "found":
+		return e.OutputData["address"].(string), nil
```

Comment:
Just as a note for possible more complicated type conversion: We can create structs and `json.Unmarshal`.

Reply:
I chose to do type conversion and added some [notes](https://github.com/mesg-foundation/core/pull/535/commits/9d4af1666fc8e4a01183bd7556747f69fb4008fe#diff-d973378190490a53fafb8156261ffeb3R35) about that. We should improve the _api_ package to accept any types otherwise using _json.Unmarshal_ for serialization can decrease the performance that can be prevented by having a better api.